### PR TITLE
add pgvecto_rs support and upgrade SQLAlchemy

### PR DESCRIPTION
**Pull request by JohnJyong**
# Description

1. support pgvecto_rs vector database
3. upgrade SQLAlchemy to version 2.0.29

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by bowenliang123** [comment]
It's great to embrace SQLAlchemy `2.x` !

File: api/core/rag/datasource/vdb/pgvecto_rs/pgvecto_rs.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        if not values['user']:
+            raise ValueError("config PGVECTO_RS_USER is required")
+        if not values['password']:
+            raise ValueError("config PGVECTO_RS_PASSWORD is required")
+        if not values['database']:
+            raise ValueError("config PGVECTO_RS_DATABASE is required")
+        return values
+
+
+class PGVectoRS(BaseVector):
```
Comment (bowenliang123) [inline api/core/rag/datasource/vdb/pgvecto_rs/pgvecto_rs.py:43]:
```suggestion
class PGVectoRSVector(BaseVector):
```

Keep the `Vector` appendix to follow the pattern like `MilvusVector`, `QdrantVector` and etc.

**Comment by bowenliang123** [comment]
Are there any planned integration tests with `pbvedto_rs` vdb?

**Comment by vaayne** [comment]
It would be better to split the pull request (PR) into two separate PRs. One for upgrading SQLAlchemy and another for adding support for `pgvector_rs`. This approach will make the process much simpler and clearer.

**Review by bowenliang123 — APPROVED** [review]
LGTM overall.
Minor comments are reported offline.

**Review by takatost — APPROVED** [review]

**Pull request merged**